packages/panna-sdk/src/react/components/buy/select-buy-provider-step.tsx

```diff
@@ -0,0 +1,168 @@
+import { Loader2Icon } from 'lucide-react';
+import { useMemo } from 'react';
+import type { UseFormReturn } from 'react-hook-form';
+import { useActiveAccount } from 'thirdweb/react';
+import { useBuyWithFiatQuotes, useSupportedTokens } from '../../hooks';
+import type { BuyWithFiatQuote } from '../../types/buy-with-fiat-quote.types';
+import { Badge } from '../ui/badge';
+import { Button } from '../ui/button';
+import { DialogHeader, DialogTitle } from '../ui/dialog';
+import { useDialogStepper } from '../ui/dialog-stepper';
+import { FormControl, FormField, FormItem, FormMessage } from '../ui/form';
+import { Typography } from '../ui/typography';
+import type { BuyFormData } from './schema';
+
+type SelectBuyProviderStepProps = {
+  form: UseFormReturn<BuyFormData>;
+};
+
+export function SelectBuyProviderStep({ form }: SelectBuyProviderStepProps) {
+  const { next } = useDialogStepper();
+
+  const activeAccount = useActiveAccount();
+  const { token, country, cryptoAmount } = form.watch();
+
+  const receiver = activeAccount?.address;
+  const { data: supportedTokens = [] } = useSupportedTokens();
+
+  const tokenAddress = useMemo(() => {
+    if (!token?.symbol) return undefined;
+    const supportedToken = supportedTokens.find(
+      (t) => t.symbol === token.symbol
+    );
+    return supportedToken?.address;
+  }, [token?.symbol, supportedTokens]);
+
+  const { data: quotes = [], isLoading } = useBuyWithFiatQuotes(
+    {
+      countryCode: country?.code || '',
+      tokenAddress,
+      cryptoAmount: cryptoAmount?.toString(),
+      receiver
+    },
+    {
+      enabled: Boolean(
+        country?.code && tokenAddress && cryptoAmount && receiver
+      )
+    }
+  );
+
+  const quotesWithBestPrice: (BuyWithFiatQuote & { best?: boolean })[] =
+    useMemo(() => {
+      const quotesWithParsedPrices = quotes.map((quote) => {
+        const numericPrice =
+          quote.price && !quote.error
+            ? parseFloat(quote.price.replace(/[^\d.-]/g, ''))
```

Comment:
What are we trying to achieve here? What is the actual data format for `quote.price`?

Please define a const with `REGEX_` and use it here. This would also help in presenting more context as to what we're trying to achieve.

Reply:
this is used to compare prices numerically to find the best (lowest) price. i made it more self explanatory